### ISSUE TITLE
[sklearn#34140] Callbacks are not left in estimator objects

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#348]
**Mirrored from [scikit-learn#34140](https://github.com/scikit-learn/scikit-learn/issues/34140)**

Original title: Callbacks are not left in estimator objects
Original labels: New Feature, Needs Decision, Callbacks
Created: 2026-05-27T19:11:41Z

---

> [!WARNING]
> This issue is not yet ready for a PR. If you are interested in contributing to scikit-learn, please have a look at our [contributing guidelines](https://scikit-learn.org/dev/developers/contributing.html), and in particular the sections for [new contributors](https://scikit-learn.org/dev/developers/contributing.html#new-contributors) and the ["Needs triage"](https://scikit-learn.org/dev/developers/contributing.html#issues-tagged-needs-triage) label.

### Describe the workflow you want to enable

In lots of other libraries like XGBoost and LightGBM, callbacks are passed as input parameters to model objects. This way, the object will contain the callbacks that were used with it - meaning that they can be created on-the-fly during construction, and retrieved from the object.

For example:
```python
from sklearn.datasets import make_regression
X, y = make_regression(random_state=123)

import xgboost as xgb
cb = xgb.callback.EarlyStopping(
    rounds=1,
    metric_name='rmse'
)
model = xgb.XGBRegressor(callbacks=[cb]).fit(X, y, eval_set=[(X, y)])
model.callbacks, cb
```
```
([<xgboost.callback.EarlyStopping at 0x7fb1f94a32c0>],
 <xgboost.callback.EarlyStopping at 0x7fb1f94a32c0>)
```

Further, some callbacks leave additional attributes in estimators, like `best_score`, and in the R interface for XGBoost, this is taken a step further by allowing each callback to leave custom public attributes under its (the callback's) given name - see example in the R documentation:
https://cran.r-project.org/web/packages/xgboost/refman/xgboost.html#xgb.Callback

But in scikit-learn, callbacks need to be kept referenced outside of the object, which is less convenient.

### Describe your proposed solution

Callbacks should leave public attributes in estimator objects after calls to `.fit()`, maybe under a common attribute `callbacks` where there would be a list with the used callbacks.

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to [70](https://github.com/godofecht/flow-scikit/issues/70) in the flow-scikit repository. Further library-specific triage and implementation will happen there.
